Description: fixed typo
Author: Scott Howard <showard@debian.org>

Index: arduino/libraries/TFT/utility/Adafruit_ST7735.h
===================================================================
--- arduino.orig/libraries/TFT/utility/Adafruit_ST7735.h	2013-06-04 13:21:52.967608085 -0400
+++ arduino/libraries/TFT/utility/Adafruit_ST7735.h	2013-06-04 14:16:25.767685867 -0400
@@ -25,7 +25,7 @@
 #else
  #include "WProgram.h"
 #endif
-#include <Adafruit_GFX.h>
+#include "Adafruit_GFX.h"
 #include <avr/pgmspace.h>
 
 // some flags for initR() :(
