Description: Use system astyle, upstream uses a custom version
Author: Scott Howard <showard@debian.org>
Index: arduino/app/src/cc/arduino/packages/formatter/AStyleInterface.java
===================================================================
--- arduino.orig/app/src/cc/arduino/packages/formatter/AStyleInterface.java	2014-04-06 19:26:14.177478383 -0400
+++ arduino/app/src/cc/arduino/packages/formatter/AStyleInterface.java	2014-04-06 19:26:14.169478383 -0400
@@ -1,4 +1,4 @@
-package cc.arduino.packages.formatter;
+//package cc.arduino.packages.formatter;
 
 import processing.app.Base;
 
Index: arduino/app/src/cc/arduino/packages/formatter/AStyle.java
===================================================================
--- arduino.orig/app/src/cc/arduino/packages/formatter/AStyle.java	2014-04-06 19:26:14.177478383 -0400
+++ arduino/app/src/cc/arduino/packages/formatter/AStyle.java	2014-04-06 19:26:14.173478383 -0400
@@ -1,4 +1,4 @@
-package cc.arduino.packages.formatter;
+//package cc.arduino.packages.formatter;
 
 import processing.app.Base;
 import processing.app.Editor;
Index: arduino/app/src/processing/app/Editor.java
===================================================================
--- arduino.orig/app/src/processing/app/Editor.java	2014-04-06 17:53:22.081505456 -0400
+++ arduino/app/src/processing/app/Editor.java	2014-04-06 19:27:57.377477881 -0400
@@ -905,7 +905,7 @@
   protected JMenu addInternalTools(JMenu menu) {
     JMenuItem item;
 
-    item = createToolMenuItem("cc.arduino.packages.formatter.AStyle");
+    item = createToolMenuItem("AStyle");
     item.setName("menuToolsAutoFormat");
     int modifiers = Toolkit.getDefaultToolkit().getMenuShortcutKeyMask();
     item.setAccelerator(KeyStroke.getKeyStroke('T', modifiers));
